openhtf/__init__.py

```diff
@@ -394,6 +394,33 @@ def with_args(self, **kwargs):
     new_info.measurements = [m.with_args(**kwargs) for m in self.measurements]
     return new_info
 
+  def with_plugs(self, **subplugs):
+    """Substitute plugs for placeholders for this phase."""
+    # Make a copy so we can have multiple of the same phase with different plugs
+    # in the same test.
+    new_info = mutablerecords.CopyRecord(self)
```

Comment:
hm, i think i have a slight preference for `new_plugs = list(self.plugs)` here, update `new_plugs` directly on line 411, and then `return mutablerecords.CopyRecord(self, plugs=new_plugs)` at the end.  It's a little clearer that plugs are the only thing you're updating.